GenericsHomework/GenericsHomework.Tests/NodeTests.cs

```diff
@@ -0,0 +1,95 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+
+namespace GenericsHomework.Tests
+{
+    [TestClass]
+    public class NodeTests
+    {
+        private static (Node<string>, Node<int>) TestNodes()
+        {
+            return (new Node<string>("The Dread Pirate Roberts"), new Node<int>(42));
+        }
+
+        [TestMethod]
+        public void Constructor_SetsCorrectValue()
+        {
+            //Assign
+            var (stringNode, intNode) = TestNodes();
+
+            //Act
+
+            //Assert
+            Assert.AreEqual<int>(42, intNode.Value);
+            Assert.AreEqual<string>("The Dread Pirate Roberts", stringNode.Value);
+        }
+
+        [TestMethod]
+        public void ToString_ReturnsCorrectString()
+        {
+            //Assign
+            var (stringNode, intNode) = TestNodes();
+
+            //Act
+
+            //Assert
+            Assert.AreEqual<string>("42", intNode.ToString());
+            Assert.AreEqual<string>("The Dread Pirate Roberts", stringNode.ToString());
+        }
+
+        [TestMethod]
+        public void NextProperty_ReturnsCorrectNode()
+        {
+            //Assign
+            Node<string> firstNode = new Node<string>("Princess Buttercup");
+            Node<string> secondNode = new Node<string>("Princess Buttercup");
+
+            //Act
+
+            //Assert
+            Assert.AreNotEqual<Node<string>>(firstNode, secondNode);
+        }
+
+        [TestMethod]
+        public void InsertMethodGivenNewNode_AssignsNextAndLoopsBackToNode()
+        {
+            //Assign
+            Node<int> headNode = new Node<int>(42);
+
+            //Act
+            headNode.Insert(43);
+
+            //Assert
+            Assert.AreEqual<string>("42", headNode.ToString());
+            Assert.AreEqual<string>("43", headNode.Next.ToString());
+            Assert.AreEqual<string>("42", headNode.Next.Next.ToString());
```

Comment:
Try to avoid using your `ToString` method when unit testing unrelated code. Your unit tests should be focused on a single _unit_ of your code, in this case, just the `Insert` method. Another way to do this below:
```suggestion
            Node<int> secondNode = headNode.Next;

            Assert.AreEqual(secondNode, headNode.Next);
            Assert.AreNotEqual(secondNode, headNode);
            Assert.AreEqual(headNode, headNode.Next.Next);
```